Add matchContents, layoutDirection, onLayoutContent, useViewportSizeMeasurement, and ignoreSafeArea support to the universal Host (#45776)
expo/expo: docs/pages/versions/unversioned/sdk/ui/universal/host.mdx

}
```

### Layout direction

Use `layoutDirection` to render the subtree as left-to-right or right-to-left. On Android and iOS, this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web, it sets the `dir` attribute on the underlying `View` so descendants inherit the chosen direction.

```tsx LayoutDirectionExample.tsx
import { Host, Row, Text } from '@expo/ui';

export default function LayoutDirectionExample() {
  return (
    <Host layoutDirection="rightToLeft">
      <Row spacing={8}>
        <Text>First</Text>
        <Text>Second</Text>
      </Row>
    </Host>
  );
}
```

### Reacting to content layout

> Use `onLayoutContent` to be notified of the current dimensions of the host's content. On Android and iOS this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web it is derived from the underlying `View`'s `onLayout` callback.

Reviewer: ```suggestion
Use `onLayoutContent` to be notified of the current dimensions of the host's content. On Android and iOS, this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web, it is derived from the underlying `View`'s `onLayout` callback.
```

```tsx OnLayoutContentExample.tsx
import { Host, Text } from '@expo/ui';

export default function OnLayoutContentExample() {
  return (
    <Host
      matchContents
      onLayoutContent={({ nativeEvent: { width, height } }) =>
        console.log(`content size: ${width}x${height}`)
      }>
      <Text>Hello, world!</Text>
    </Host>
  );
}
```

### Filling the viewport

> Use `useViewportSizeMeasurement` for content that should size to the available viewport space. On Android and iOS this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web the host's underlying `View` is given the current window's width and height; any explicit `style` you pass still wins.

Reviewer: ```suggestion
Use `useViewportSizeMeasurement` for content that should size to the available viewport space. On Android and iOS, this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web, the host's underlying `View` is given the current window's width and height; any explicit `style` you pass still wins.
```

```tsx UseViewportSizeMeasurementExample.tsx
import { Host, Column, Text } from '@expo/ui';

export default function UseViewportSizeMeasurementExample() {
  return (
    <Host useViewportSizeMeasurement>
      <Column spacing={12} alignment="center">
        <Text>Fills the viewport</Text>
      </Column>
    </Host>
  );
}
```

### Ignoring safe areas

> By default, `Host` respects the device safe area insets (notch, home indicator, etc.). Use `ignoreSafeArea="all"` to let content extend edge-to-edge. On Android and iOS this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web it is implemented via the CSS `env(safe-area-inset-*)` values applied as padding on the underlying `View`; `"keyboard"` is treated the same as the default since the browser's `env()` function does not expose keyboard insets.

Reviewer: ```suggestion
By default, `Host` respects the device safe area insets (notch, home indicator, an so on). Use `ignoreSafeArea="all"` to let content extend edge-to-edge. On Android and iOS, this is forwarded to the platform-native `Host` (see [Jetpack Compose](../jetpack-compose/host)/[SwiftUI](../swift-ui/host) for the exact platform semantics). On web, it is implemented via the CSS `env(safe-area-inset-*)` values applied as padding on the underlying `View`. `"keyboard"` is treated the same as the default since the browser's `env()` function does not expose keyboard insets.
```
